numpy/core/src/multiarray/descriptor.c

```diff
@@ -1091,7 +1087,7 @@ _convert_from_dict(PyObject *obj, int align)
     totalsize = 0;
     for (i = 0; i < n; i++) {
         PyObject *tup, *descr, *ind, *title, *name, *off;
-        int len, ret, _align = 1;
+        int len, _align = 1;
```

Comment:
Mystery solved, this `ret` was being used uninitialized further down the file, in places where it made no sense to check it anyway.

Reply:
Odd that the compiler warning tests did not find that :(.

Reply:
Well, there was a codepath that set it, so the error was likely something like "maybe uninitialized", rather than "is uninitialized".